ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/activity/PaywallActivityLauncher.kt

```diff
@@ -183,4 +187,10 @@ class PaywallActivityLauncher(resultCaller: ActivityResultCaller, resultHandler:
             }
         }
     }
+
+    private fun isActivityFinishing(): Boolean {
+        val activity = weakActivity.get()
+        val fragment = weakFragment.get()
+        return activity?.isFinishing == true || fragment?.activity?.isFinishing == true
+    }
```

Comment:
Maybe this should also return true if the Activity is completely dereferenced? That is, if `activity` or `fragment.activity` is null (but not if only the `fragment` is null imo). What do you think?

Reply:
Yeah absolutely! Fixing this, thanks!

Reply:
Ah I remember now... The reason I didn't do that was because a developer could potentially implement their own `ActivityResultCaller` which is not an activity or a fragment (honestly I don't know of any valid use cases, but they could do this), and we would still want to display the paywall there, or it could potentially cause issues in that case... 

I think we should leave as is, and if we don't have an activity to check, just continue as we do right now. Lmk what you think!

Reply:
I agree that this is probably good enough. If we really want to catch the case where the `ActivityResultCaller` is neither an Activity nor a Fragment, we can store that info when we assign the WeakReferences. Not a blocker for me.